dependencies/dependencies.go

```diff
@@ -12,40 +13,47 @@ const InterpreterDepsKey = "interpreter"
 type Interface interface {
 	HTTPClient() (*http.Client, error)
 	SecretService() (SecretService, error)
+	URLValidator() (url.Validator, error)
 }
 
 type DefaultDependencies struct {
-	httpclient    *http.Client
-	secretservice SecretService
+	Deps Deps
```

Comment:
maybe I'm overthinking the design but if `deps` are shared between components they should be implemented as immutable.  

I don't mind double-encapsulating deps to do this, but making deps a private member of DefaultDependencies.  Then we can do `NewDefaultDependencies(deps)`  to construct it.